src/pipeline_migration/actions/migrate.py

```diff
@@ -101,6 +103,65 @@ class InvalidRenovateUpgradesData(ValueError):
     """Raise this error if any required data is missing in the given Renovate upgrades"""
 
 
+def only_tags_pinned_by_version_revision(tags_info: Iterable[dict]) -> Generator[dict, Any, None]:
+    regex = re.compile(TASK_TAG_REGEXP)
+    for tag_info in tags_info:
+        if regex.match(tag_info["name"]):
+            yield tag_info
+
+
+def filter_out_bundles_built_from_older_version(
+    tags_info: Iterable[dict], stop_at_digest: str
+) -> Generator[dict, Any, None]:
+    """Filter out bundles built from older versions
+
+    Generally, once a new version is bumped for a task, there should be no new
+    bundle build for the older versions. However, the corner case happened.
+    Refer to STONEBLD-3667.
```

Comment:
> Generally, once a new version is bumped for a task, there should be no new bundle build for the older versions.

This is not really true. New version being available doesn't automatically mean the old one is deprecated. Currently, deprecation is a manual process that  _requires_ a new bundle to be built for the old task. See https://github.com/konflux-ci/build-definitions?tab=readme-ov-file#task-deprecation

TLDR: it's a fairly common case, not a corner case

Reply:
Then, I think it can be reworded as:

Once a new version is bumped for a task, there should be no new bundle builds to introduce migrations.

Reply:
I think that's fair. Technically, nothing prevents that from happening, but it should be an unlikely use case